### PR TITLE
Filter fix

### LINKED ISSUE #117: Odd data_extract_spec behaviour
I'm running the roxygen example of tm_t_tte (on main) in tmc and it has:

```
paramcd = choices_selected(
  value_choices(ADTTE, "PARAMCD", "PARAM"),
  "OS"
),
```

And when I load the app it  does have "OS" selected but then immediately replaces it with CRSD (the first option in the dropdown list)

Does anyone know why? Is this behaviour widespread? Do we care?

I suspect it's something buried deep in teal.transform (I mention because when the original "OS" is selected the shinyvalidate messages aren't happy (though after the initial load is completed they then behave correctly) - interestingly the first pass through (with OS) correctly shows the column labels (OS 'Overall Survival') but then when it overwrites to CRSD it doesn't...

Transfer to tmc if that makes sense

### PR DESCRIPTION
**Pull request by mhallal1**
closes #117 

The two `updateOptionalSelectInput` could not be removed because of their need when the selection is not assigned by default, check `tm_g_distribution` example app from tmg for clarification.

We now account for when something is pre-selected in `data_extract_filter_srv` and assign the selected value to it.
Test with app below and try to remove the `selected` value also:
Also test with modules from tmg and tmc.
```
library(shiny)

extract_ui <- function(id, data_extract) {
  ns <- NS(id)
  teal.widgets::standard_layout(
    output = teal.widgets::white_small_well(verbatimTextOutput(ns("output"))),
    encoding = data_extract_ui(ns("data_extract"), label = "variable", data_extract)
  )
}

extract_srv <- function(id, data, data_extract) {
  moduleServer(id, function(input, output, session) {
    reactive_extract_input <- data_extract_srv("data_extract", data, data_extract, join_keys = attr(data, "join_keys"))

    merged <- merge_expression_srv(selector_list = reactive(list(x = reactive_extract_input)), datasets = data, join_keys = attr(data, "join_keys"))
    s <- reactive(merged()$expr)
    output$output <- renderPrint(s())
  })
}

tm_extract <- function(label = "extract", data_extract) {
  teal::module(
    label = label,
    server = extract_srv,
    ui = extract_ui,
    server_args = list(data_extract = data_extract),
    ui_args = list(data_extract = data_extract)
  )
}

# Define data.frame objects
ADSL <- scda::synthetic_cdisc_data("latest")$adsl # nolint
ADTTE <- scda::synthetic_cdisc_data("latest")$adtte # nolint

# create a list of data.frame objects
data <- cdisc_data(
  cdisc_dataset("ADSL", ADSL),
  cdisc_dataset("ADTTE", ADTTE)
)

app <- teal::init(
  data = data,
  modules = list(
    tm_extract(
      "extract-4",
      data_extract_spec(
        dataname = "ADTTE",
        filter = filter_spec(
          vars = "PARAMCD",
          value_choices(ADTTE, "PARAMCD", "PARAM"), 
          selected = "OS"
        )
      )
    )
  )
)

runApp(app)
```

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -60,4 +60,4 @@ Encoding: UTF-8
 Language: en-US
 LazyData: true
 Roxygen: list(markdown = TRUE)
-RoxygenNote: 7.2.1
+RoxygenNote: 7.2.3
```
Comment (kpagacz) [inline DESCRIPTION:63]: Intended?

File: R/data_extract_filter_module.R [hunk 2]
```diff
@@ -78,10 +78,14 @@ data_extract_filter_srv <- function(id, datasets, filter) {
...
-              selected <- choices
+            selected <- if (!is.null(filter$selected)) {
+              filter$selected
             } else {
-              selected <- choices[1]
+              if (filter$multiple) {
+                choices
+              } else {
+                choices[1]
+              }
```
Comment (kpagacz) [inline R/data_extract_filter_module.R:88]: God, the ifs xd

Comment (kpagacz) [inline R/data_extract_filter_module.R:88]: What's the type of `choices` in here? Is it possible to abuse this `R` thing:
```r
val = 2
print(val[1])
# prints 2
```

Comment (kpagacz) [reply R/data_extract_filter_module.R:88]: Weirdly enough, I have a vague recollection of asking myself this question in these parts of the code in the past xd

**Review by asbates — APPROVED** [review]
Looks like issue is resolved to me. There may be another issue but it's not directly related to #117 so I opened #120.

File: DESCRIPTION (hunk 1 above)
Comment (mhallal1) [reply DESCRIPTION:63]: yes.

File: R/data_extract_filter_module.R (hunk 2 above)
Comment (mhallal1) [reply R/data_extract_filter_module.R:88]: I tested `choices` with `AGE` and `SEX` which returned the types `integer` and `character` respectively.

**Review by gogonzo — APPROVED** [review]
👍

**Pull request merged**